pdns/recursordist/syncres.cc

```diff
@@ -4439,6 +4441,12 @@ void SyncRes::sanitizeRecordsPass2(const std::string& prefix, LWResult& lwr, con
         LOG(prefix << qname << ": Removing irrelevent record '" << rec->toString() << "' in the ANSWER section received from " << auth << endl);
         skipvec[counter] = true;
         ++skipCount;
+      }
+      // If we have a CNAME, skip other records types for that name except RRSIGs
+      if (cnamesSeen.find(rec->d_name) != cnamesSeen.end() && rec->d_type != QType::CNAME && rec->d_type != QType::RRSIG) {
```

Comment:
Even `DNAME`?

Reply:
Yes, rethinking this, the PR is too strict. It's probably better to only remove records with the requested qname and type in the auth's answer if a CNAME for qname is present.